test/units/galaxy/test_collection_install.py

```diff
@@ -434,6 +434,58 @@ def test_build_requirement_from_tar_invalid_manifest(tmp_path_factory):
+    tmp_path = os.path.join(os.path.split(collection_artifact[1])[0], b'temp')
+    concrete_artifact_cm = collection.concrete_artifact_manager.ConcreteArtifactsManager(tmp_path, validate_certs=False)
+    expected = (
+        r'The following {collection_name!s} collection requirement entry keys are not '
+        r'valid: invalid_key, key'.format(collection_name=to_text(collection_artifact[1]))
+    )
+    with pytest.raises(AnsibleError, match=expected):
+        Requirement.from_requirement_dict({'name': to_text(collection_artifact[1]), 'invalid_key': 'foo', 'key': 'bar'}, concrete_artifact_cm)
+
+
+def test_build_requirement_from_tar_invalid_name_with_source(collection_artifact):
+    tmp_path = os.path.join(os.path.split(collection_artifact[1])[0], b'temp')
+    concrete_artifact_cm = collection.concrete_artifact_manager.ConcreteArtifactsManager(tmp_path, validate_certs=False)
+    expected = (
+        r"The foo 'source' key is not applicable when 'name' is not a FQCN\."
+    )
+    with pytest.raises(AnsibleError, match=expected):
+        Requirement.from_requirement_dict({'name': 'foo', 'source': 'foobar'}, concrete_artifact_cm)
+
+
+def test_build_requirement_from_tar_invalid_semantic_version(collection_artifact):
+    tmp_path = os.path.join(os.path.split(collection_artifact[1])[0], b'temp')
+    concrete_artifact_cm = collection.concrete_artifact_manager.ConcreteArtifactsManager(tmp_path, validate_certs=False)
+    expected = (
+        r"The foo.bar 'version' key must be a valid collection version "
+        r'\(see specification at https://semver\.org/\)\.'
+    )
+    with pytest.raises(AnsibleError, match=expected):
+        Requirement.from_requirement_dict({'name': 'foo.bar', 'version': 'foobar'}, concrete_artifact_cm)
+
+
+def test_build_requirement_from_tar_invalid_semantic_version_specifier(collection_artifact):
+    tmp_path = os.path.join(os.path.split(collection_artifact[1])[0], b'temp')
+    concrete_artifact_cm = collection.concrete_artifact_manager.ConcreteArtifactsManager(tmp_path, validate_certs=False)
+    expected = (
+        r"The foo.bar 'version' key must be a valid collection version "
+        r'\(see specification at https://semver\.org/\)\.'
+    )
+    with pytest.raises(AnsibleError, match=expected):
+        Requirement.from_requirement_dict({'name': 'foo.bar', 'version': '>=foobar'}, concrete_artifact_cm)
+
+
+def test_build_requirement_from_tar_pre_release_version(collection_artifact):
+    tmp_path = os.path.join(os.path.split(collection_artifact[1])[0], b'temp')
+    concrete_artifact_cm = collection.concrete_artifact_manager.ConcreteArtifactsManager(tmp_path, validate_certs=False)
+    try:
+        Requirement.from_requirement_dict({'name': 'foo.bar', 'version': '=2.4.5-rc5'}, concrete_artifact_cm)
+    except AnsibleError as err:
+        pytest.fail(f'Failed to handle a collection with a pre-release version: {err}')
```

Comment:
I don't think that this try/except dance is necessary. Pytest would surface the exception regardless. It'd be enough to add a code comment that this is a smoke test checking for no exceptions.